### PR TITLE
Fix attribs

### PR DESCRIPTION
**Pull request by hoshiryu**
* **Please check if the PR fulfills these requirements**
- [ ] Is the Pull-Request done against the right branch:
  - `master-v1`: for changes related to the Radium Stable Release V1.
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

Be aware that the PR request cannot be accepted if it doesn't pass the Continuous Integration tests.

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
This PR fixes some issues with the management of mesh attributes.

* **What is the current behavior?** (You can also link to an open issue here)
- When removing an attribute from a mesh though its handle, we get the attribute through `getAttrib`, and then use it's name to look for it to check if it exists, which causes a crash when it does not.
- When building a `TopologicalMesh`, we copy all the mesh attributes onto halfedges, but without any check on the data size, thus causing a crash when the attribute is empty.
- Upon initialization, the `SkinningComponent` ask for a UV attribute within the mesh, but leave it empty if it does not exist yet, thus putting the mesh in an inconsistent state.

* **What is the new behavior (if this is a feature change)?**
- When removing an attribute, we look for the attribute using its name within the handle.
- When copying the attributes, we first check the attribute has exactly one value per vertex in the input `TriangleMesh`, and output a warning message otherwise.
- The `SkinningComponent` now adds the UV attribute only when showing the skinning weights.
When disabling weights display, the attribute is reset to its initial state (if present from the beginning) or removed (otherwise).

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:
- The method `getSize` from `Attrib<T>` previously returned the dimension of `T`. It now returns the number of elements in the data container, which seams more suited with its name.
- Fix Debug mode compilation.
- Prevent building of glm tests and static libraries.

### CODE REVIEW

File: Plugins/Skinning/src/SkinningComponent.cpp [hunk 1]
```diff
@@ -429,17 +434,28 @@ void SkinningComponent::showWeights( bool on ) {
...
-        mesh->getAttrib( handle ).data() = m_baseUV;
+        handle = mesh->getAttribHandle<Ra::Core::Vector3>( attrUV );
+        // if the UV attrib existed before, reset it, otherwise remove it.
+        if ( m_baseUV.size() > 0 )
+        {
+            mesh->getAttrib( handle ).data() = m_baseUV;
+        } else
+        {
+            mesh->removeAttrib( handle );
+            m_meshWritter(); // FIXME: update Engine::Mesh's attrib handles
```
Comment (nmellado) [inline Plugins/Skinning/src/SkinningComponent.cpp]: check without

File: src/Core/Geometry/TopologicalMesh.cpp [hunk 2]
```diff
@@ -23,10 +23,17 @@ void addAttribPairToTopo( const TriangleMesh& triMesh, TopologicalMesh* topoMesh
                           AttribManager::value_type attr, std::vector<PropPair<T>>& vprop,
                           std::vector<OpenMesh::HPropHandleT<T>>& pph ) {
     AttribHandle<T> h = triMesh.getAttribHandle<T>( attr->getName() );
-    OpenMesh::HPropHandleT<T> oh;
-    topoMesh->add_property( oh, attr->getName() );
-    vprop.push_back( std::make_pair( h, oh ) );
-    pph.push_back( oh );
+    if ( attr->getSize() == triMesh.vertices().size() )
+    {
+        OpenMesh::HPropHandleT<T> oh;
```
Comment (nmellado) [inline src/Core/Geometry/TopologicalMesh.cpp]: add test

File: src/Core/Utils/Attribs.hpp [hunk 3]
```diff
@@ -27,7 +27,7 @@ class AttribBase {
     void setName( std::string name ) { m_name = name; }
     virtual void resize( size_t s ) = 0;
 
-    virtual uint getSize() = 0;
+    virtual size_t getSize() = 0;
```
Comment (nmellado) [inline src/Core/Utils/Attribs.hpp]: add documentation

File: src/Core/Utils/Attribs.hpp [hunk 4]
```diff
@@ -68,7 +68,7 @@ class Attrib : public AttribBase {
     inline const Container& data() const { return m_data; }
 
     virtual ~Attrib() { m_data.clear(); }
-    uint getSize() override { return Container::Matrix::RowsAtCompileTime; }
+    size_t getSize() override { return m_data.size(); }
     int getStride() override { return sizeof( typename Container::value_type ); }
```
Comment (nmellado) [inline src/Core/Utils/Attribs.hpp]: add warning: does not work for dynamic and sparse matrices

File: Plugins/Skinning/src/SkinningComponent.cpp (hunk 1 above)
Comment (hoshiryu) [reply Plugins/Skinning/src/SkinningComponent.cpp]: works fine without

Comment (nmellado) [reply Plugins/Skinning/src/SkinningComponent.cpp]: Great, please remove the comment then.

File: src/Core/Utils/Attribs.hpp (hunk 3 above)
Comment (dlyr) [reply src/Core/Utils/Attribs.hpp]: well this one is to be used along with stride to send data to the gpu. 
Actually it is not used yet ... (size is getted from VecArray::Vector::RowsAtCompileTime with the sendGL method of Engine::Mesh).

Nevertheless we could imagine having both meth (get size : number of attributes and get Whatever : dim of each value).

But this function is clearly related to gpu stuff, and size should always be the number of vertices in the mesh (so user don't have to worry about one attrib specific size).

File: Plugins/Skinning/src/SkinningComponent.cpp (hunk 1 above)
Comment (hoshiryu) [reply Plugins/Skinning/src/SkinningComponent.cpp]: Actually need it.

File: src/Core/Utils/Attribs.hpp [hunk 5]
```diff
@@ -19,58 +19,109 @@ namespace Utils {
...
     virtual void resize( size_t s ) = 0;
 
-    virtual uint getSize() = 0;
+    /**
+     * Return the number of elements in the attribute content.
+     */
+    virtual size_t getSize() = 0;
+
+    /**
+     * Return the size, in bytes, of one attribute element.
```
Comment (dlyr) [inline src/Core/Utils/Attribs.hpp]: Well stide is the offset, in byte, to go from one attirbute address to ne next one.
As defined here https://www.khronos.org/opengl/wiki/Vertex_Specification

Comment (MathiasPaulin) [inline src/Core/Utils/Attribs.hpp]:
```suggestion
     * Return the stride, in bytes, from one attribute address to the next one.
```

**Pull request merged**